src/tox_ansible/tox_helper.py

```diff
@@ -94,6 +94,8 @@ def add_envconfigs(self, tox_cases, options):
                 self.config, tox_case.get_name(), section, reader._subs, self.config
             )
             config.tox_case = tox_case
+            # We do not want to create new environments for each command we run
+            config.envdir = py._path.local.LocalPath(abspath(join(".tox", "ansible")))
```

Comment:
Does this play nicely with running tox from a subfolder of the {toxinidir} with the other change you made?

Also, I considered setting this to be equal to .tox/{driver} as I often will run in parallel with different drivers.

Reply:
TBH, I am quite sure molecule will choke if you want to run same scenario with multiple drivers in parallel. I personally do not see a good reason to use drivers as a env-matrix factor, we can just install all of them in the same environment.

On the other hand, I am almost sure this one does not play well with the other factors: python versions and ansible versions, likely that now they all run using unexpected versions of python or ansible.

If confirmed we need to assure that expanded environments use these facts when producing the env name, so we would have something like "py27-ansible29" instead of just ansible.

I also need to double check the sub-directory calling, based on my tests it worked.

Reply:
The issue isn't running molecule with multiple drivers in parallel. It's tox. Let's say I have two scenarios and they generate tox environments

foo-podman
foo-openstack

If I invoke with `tox -p`, it will attempt to run both of these at once. That's fine, the Podman instance will run locally and consume my resources while the OpenStack instance will run remotely and consume someone else's resources.

Unfortunately, tox picks up the list of dependencies as ['ansible', 'molecule', 'molecule-podman'] and ['ansible', 'molecule', 'molecule-openstack'] and it attempts to create two different environments in the same path simultaneously. That's bad, especially for when the parallel run contains many environments and one "pulls the rug out from under" another. I've had it happen before in my testing for different ways to address this conundrum using exactly the above solution. These interleaving tasks occur:

0. Podman environment is created with appropriate pip deps
1. Podman scenario runs create
2. Podman scenario runs converge
3. OpenStack virtualenv is create with different pip deps - tox sees they're different and deletes the Podman virtualenv to replace it with the OpenStack one
4. At some later point Podman scenario tries to run destroy - this fails because molecule-podman is no longer found in the virtualenv.

At the very least we need to have .tox/{driver} and, as you point out, we need to have it be .tox/{python}-{ansible}-{driver}. And add an exception for when a requirements.txt file is auto-discovered in the molecule scenario, as that will result in a completely different set of deps.